src/pset/mod.rs

```diff
@@ -248,13 +253,296 @@ impl PartiallySignedTransaction {
+    /// blinding inputs and scalars from [`Global`] scalars field.
+    /// `inp_secrets` and `out_secrets` must be consistent by [`Output`] `blinder_index` field
+    /// For each output, the corresponding inp_secrets\[out.blinder_index\] must be present
+    /// # Parameters
+    ///
+    /// * `inp_secrets`: [`TxOutSecrets`] corresponding to owned inputs. Use [`None`] for non-owned outputs
+    ///
+    pub fn blind_last<C: secp256k1_zkp::Signing, R: RngCore + CryptoRng>(
+        &mut self,
+        rng: &mut R,
+        secp: &secp256k1_zkp::Secp256k1<C>,
+        inp_txout_sec: &[Option<&TxOutSecrets>],
+    ) -> Result<(), PsetBlindError> {
+        let (_inp_secrets, mut outs_to_blind) = self.blind_checks(inp_txout_sec)?;
+
+        // If there are more than 1 outs to blind
+        // blind all outs but the last one
+        let last_out_index = outs_to_blind.pop().unwrap();
+        if !outs_to_blind.is_empty() {
+            // Don't blind the last output
+            self.outputs[last_out_index].blinder_index = None;
+            self.blind_non_last(rng, secp, inp_txout_sec)?;
+        }
+        // blind the last txout
+        let asset = self.outputs[last_out_index].asset.explicit().ok_or(PsetBlindError::MustHaveExplicitTxOut(last_out_index))?;
+        let value = self.outputs[last_out_index].amount.explicit().ok_or(PsetBlindError::MustHaveExplicitTxOut(last_out_index))?;
+        let out_abf = AssetBlindingFactor::new(rng);
+        let out_asset = confidential::Asset::new_confidential(secp, asset, out_abf);
+        let out_asset_commitment = out_asset.commitment().expect("confidential asset");
+
+        // Get all the explicit outputs
+        let mut exp_out_secrets = vec![];
+        for (i, out) in self.outputs.iter().enumerate() {
+            if out.blinding_key.is_none() {
+                let amt = out.amount.explicit().ok_or(PsetBlindError::MustHaveExplicitTxOut(i))?;
+                exp_out_secrets.push((amt, AssetBlindingFactor::zero(), ValueBlindingFactor::zero()));
+            }
+        }
+        let vbf = ValueBlindingFactor::last(
+            secp,
+            value,
+            out_abf,
+            &[],
+            &exp_out_secrets,
+        );
+
+        let mut final_blind =
+            SecretKey::from_slice(vbf.into_inner().as_ref()).expect("Non-zero random value");
+
+        // Add all the scalars
+        for value_diff in self.global.scalars.iter() {
+            if value_diff.as_ref() == &[0u8; 32] {
+                continue;
+            } else {
+                // Valid secret key
+                let scalar =
+                    SecretKey::from_slice(value_diff.as_ref()).expect("Valid non-zero tweak");
+                final_blind
+                    .add_assign(scalar.as_ref())
+                    .expect("Valid tweak");
+            }
+        }
+        // The last vbf
+        let out_vbf = ValueBlindingFactor::from_slice(final_blind.as_ref()).expect("32 byte");
```

Comment:
Can we implement `add_assign` directly on `ValueBlindingFactor`, and/or implement a conversion funcion from `SecretKey`? It would eliminate a lot of this awkwardness.